packages/web/lib/hooks/useCalendar.ts

```diff
@@ -71,15 +74,66 @@ export const useCalendar = (clamp?: number): UseCalendarReturnTypes => {
   });
   const [limit, setLimit] = useState<number>(clamp || 0);
   const [fetching, setFetching] = useState(false);
-  const [error, setError] = useState<Error | undefined>(undefined);
+  const [error, setError] = useState<Error>();
   const calendarICS = `https://calendar.google.com/calendar/ical/${calendarId}%40group.calendar.google.com/public/basic.ics`;
 
-  // strip out the +++cover+++ from the description
-  const cleanDescription = (desc: string) =>
-    desc ? desc.replace(/(\+\+\+).*(\+\+\+)/, '') : '';
+  /**
+   * sanitize the description & get the cover image url if there is one and return the url & sanitized description
+   */
+  const cleanDescription = (
+    desc: string,
+  ): { cover?: string; description: string; originalDesc?: string } => {
+    if (!desc) {
+      return {
+        cover: undefined,
+        description: '',
+        originalDesc: desc,
+      };
+    }
+
+    function extractCoverUrl(input: string) {
+      const splitInput = input.split('+++<br>');
+      const coverSection = splitInput.length > 1 ? splitInput[1] : null;
+      const urlRegex = /(https?:\/\/[^"\s<]+)/g;
+
+      let coverUrl;
+      const modifiedInput = splitInput[0];
+
+      if (coverSection) {
+        const urls = coverSection.match(urlRegex);
+        if (urls && urls.length) {
+          coverUrl = urls[urls.length - 1];
+        }
+      }
+
+      return {
+        coverUrl,
+        modifiedInput,
+      };
+    }
+
+    const { coverUrl, modifiedInput } = extractCoverUrl(desc);
+
+    let cleanedDesc: string;
+
+    cleanedDesc = modifiedInput.replace(
+      /(^(?:<br>)+|(?:<br>)+$|(?:<br>){2,})|(?:<\w+><\/\w+>\s*)+/g,
```

Comment:
> Do you want to replace repeated newlines with no newlines or with a single newline?

Ideally, if there's one or two new lines, keep them, otherwise remove. 
 
> Also, you might use a backreference on your tag matching, to prevent partially matching something like `<b><br></b>`. It would be `<(\w+)[^>]*><\/\1>`.

Cool, thanks. 
 
> What tag are you seeing inserted with no content anyway? _(I don't have the private key to pull actual data.)_

There is very often empty `<ul>` and sometimes `<span>` in the strings. The output from the events description is awful. I've sent you info on obtaining the env vars via email & discord.